pdns/dnsdistdist/dnsdist-lua-ffi.cc

```diff
@@ -598,3 +598,33 @@ void setupLuaFFIPerThreadContext(LuaContext& luaCtx)
   luaCtx.executeCode(getLuaFFIWrappers());
 #endif
 }
+
+size_t dnsdist_ffi_generate_proxy_protocol_payload(const size_t addrSize, const void* srcAddr, const void* dstAddr, const uint16_t srcPort, const uint16_t dstPort, const bool tcp, const size_t valuesCount, const dnsdist_ffi_proxy_protocol_value* values, void* out, const size_t outSize)
+{
+  ComboAddress src, dst;
+  if (addrSize != sizeof(src.sin4.sin_addr) && addrSize != sizeof(src.sin6.sin6_addr.s6_addr)) {
+    return 0;
+  }
+
+  src = makeComboAddressFromRaw(addrSize == sizeof(src.sin4.sin_addr) ? 4 : 6, reinterpret_cast<const char*>(srcAddr), addrSize);
```

Comment:
Just wondering if we want to handle the potential exception that may be thrown here ?

Reply:
We do! Because throwing an exception from the FFI interface does not usually end well.